### PR TITLE
Collection contents locking

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/998

### CODE REVIEW

File: src/org/exist/util/io/Resource.java [hunk 1]
```diff
@@ -852,14 +852,15 @@ public Collection getCollection() throws IOException {
...
+                    final List<String> list = new ArrayList<>();
                     for (final CollectionEntry entry : collection.getEntries(broker)) {
                         list.add(entry.getUri().lastSegment().toString());
                     }
 
                     return list.toArray(new String[list.size()]);
                 }
-    		} catch (final PermissionDeniedException | EXistException e) {
+    		} catch (final LockException | PermissionDeniedException | EXistException e) {
+                LOG.error(e);
```
Comment (dizzzz) [inline src/org/exist/util/io/Resource.java]: yep, looks like a good idea

File: src/org/exist/backup/SystemExport.java [hunk 2]
```diff
@@ -918,8 +919,8 @@ public void startElement( String uri, String localName, String qName, Attributes
                             serializer.endElement( Namespaces.EXIST_NS, "deleted", "deleted" );
                         }
                     }
-                } catch(final PermissionDeniedException pde) {
-                    throw new SAXException("Unable to process :" + qName + ": " + pde.getMessage(), pde);
+                } catch(final LockException | PermissionDeniedException e) {
```
Comment (dizzzz) [inline src/org/exist/backup/SystemExport.java]: somehow I'd expect that more code would need to have this additional catch; clearly this is not the case. Interesting....

**Comment by dizzzz** [comment]
@wolfgangmm @shabanovd  ?
@ljo Does this hurt your current testing? (I don't expect so....)

**Comment by shabanovd** [comment]
lgtm, during review [that bug](https://github.com/eXist-db/exist/pull/1000) was found

**Comment by adamretter** [comment]
@shabanovd Good enough to merge?

**Comment by dizzzz** [comment]
oops.. /home/travis/build/eXist-db/exist/build/scripts/junit.xml:125: Test org.exist.xslt.TransfomerFactoryAllocatorTest failed

**Comment by adamretter** [comment]
@dizzzz That was introduced by Dmitriy's removal of XSLT, I just sent a seperate PR for that

**Comment by shabanovd** [comment]
that potentially can lead to deadlock, but it's right way move. If no objection I can merge @wolfgangmm @ljo

**Comment by adamretter** [comment]
@shabanovd Where is the potential deadlock?

**Comment by shabanovd** [comment]
@adamretter when system have a lot of lock each code modification have "potential deadlock"

**Comment by adamretter** [comment]
@shabanovd I am afraid that doesn't tell me anything apart from "Some locking scenarios can lead to deadlocks", which is the case with any complex locking system.

**Comment by shabanovd** [comment]
@adamretter I didn't say that I can reproduce deadlock, just speak up my fears and optimism

**Comment by adamretter** [comment]
@shabanovd Okay well help me out then buddy and merge it ;-)

**Comment by shabanovd** [comment]
@adamretter I'm waiting for @ljo and @wolfgangmm answers on questions, I did not run any tests just did code review

**Comment by ljo** [comment]
Ok. Please go ahead.

**Pull request merged**